proc/switchproc/switch.go

```diff
@@ -0,0 +1,166 @@
+package switchproc
```

Comment:
Maybe call this switcher so it matches the name of the type returned by New?

Reply:
The convention in the proc packages has been 
```
package foo 

type Proc struct {
...
}

func New() *Proc {
}
```
So, foo.New() creates a *foo.Proc

Reply:
@mccanne @nwt two things at play here:

- re the convention on package name, this package should be called `switch`, but I had to call it `switchproc` due to `switch` being a reserved word.
- re the `Switcher` type, this is modelled on the existing `split.go` file, which has a `type Splitter`. There is a `type Proc` below in this file, just like in `split.go`.

Reply:
my bad

Reply:
I think it's confusing for `switchproc.New` to return a `Switcher`, and I think we should fix that, either by changing the package name to `switcher` or changing the constructor name to `NewSwitcher`, though I'm fine saving the change for another PR.

What's going on in split.go is also confusing, and I think we should also fix that in a separate PR.

Reply:
@nwt if we're going down this path... I think naming-wise this should all derive from the AST names, where we have SwitchProc (like PArallelProc, SequentialProc, etc). Assuming those AST names are staying, then I would keep the package name `switchproc` and rename type Switcher to Proc (per @mccanne's point above). What do you think?

Reply:
Sorry for my tardy response!

Renaming `Switcher` to `Proc` doesn't feel quite right because it doesn't implement `proc.Interface`, which is a characteristic of all the other types named `Proc`.

Reply:
@nwt ok - done in b6c7e3bfe9fbb9db8f7508ee70ff6b757cbf5bf2